OAuth2/Client/Impl/YahooClient.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+
+using Newtonsoft.Json.Linq;
+using OAuth2.Configuration;
+using OAuth2.Infrastructure;
+using OAuth2.Models;
+using RestSharp.Authenticators;
+
+namespace OAuth2.Client.Impl
+{
+    /// <summary>
+    /// Yahoo client 
+    /// Right now only Yahoo Gemini and Yahoo Social support OAuth2
+    /// https://developer.yahoo.com/oauth2/guide/
+    /// </summary>
+    public class YahooClient : OAuth2Client
+    {
+        public string userProfileGUID;
```

Comment:
Pull request looks great, but could you please leave this field private and provide access to its value via property with getter?